src/ChanperfTracker.ts

```diff
@@ -0,0 +1,156 @@
+                this.chanperfHistory.anonMemKib = parseInt(anonMemKib);
+                this.chanperfHistory.fileMemKib = parseInt(fileMemKib);
+                this.chanperfHistory.sharedMemKib = parseInt(sharedMemKib);
+                this.chanperfHistory.totalCpuUsage = parseInt(totalCpuUsage) > 100 ? 100 : parseInt(totalCpuUsage);
+                this.chanperfHistory.userCpuUsage = parseInt(userCpuUsage) > 100 ? 100 : parseInt(userCpuUsage);
+                this.chanperfHistory.sysCpuUsage = parseInt(sysCpuUsage) > 100 ? 100 : parseInt(sysCpuUsage);
+
+                this.chanperfHistory.dataSet.totalMemKib.push(parseInt(totalMemKib));
+                this.chanperfHistory.dataSet.anonMemKib.push(parseInt(anonMemKib));
+                this.chanperfHistory.dataSet.fileMemKib.push(parseInt(fileMemKib));
+                this.chanperfHistory.dataSet.sharedMemKib.push(parseInt(sharedMemKib));
+                this.chanperfHistory.dataSet.totalCpuUsage.push(parseInt(totalCpuUsage) > 100 ? 100 : parseInt(totalCpuUsage));
+                this.chanperfHistory.dataSet.userCpuUsage.push(parseInt(userCpuUsage) > 100 ? 100 : parseInt(userCpuUsage));
+                this.chanperfHistory.dataSet.sysCpuUsage.push(parseInt(sysCpuUsage) > 100 ? 100 : parseInt(sysCpuUsage));
+
+                if (!this.filterOutLogs) {
+                    normalOutput += line + '\n';
+                }
+                dataChanged = true;
+            } else {
+                // see the following for an explanation for this regex: https://regex101.com/r/Nwqd5e/1/
+                let noInfoAvailableMatch = /channel: *(mem *and *cpu *data *not *available)/gim.exec(line);
+
+                if (noInfoAvailableMatch) {
+                    this.chanperfHistory.missingInfoMessage = noInfoAvailableMatch[1];
+
+                    if (!this.filterOutLogs) {
+                        normalOutput += line + '\n';
+                    }
+                    dataChanged = true;
+                } else if (line) {
+                    normalOutput += line + '\n';
+                }
+            }
+
+        }
+
+        if (dataChanged) {
+            this.emit('chanperf-event', this.chanperfHistory);
+        }
+
+        return normalOutput;
+    }
+
+    /**
+     * Helper function used to create a new ChanperfHistory object with default values
+     */
+    private createNewChanperfHistory(): ChanperfHistory {
+        return {
+            totalMemKib: 0,
+            anonMemKib: 0,
+            fileMemKib: 0,
+            sharedMemKib: 0,
+            totalCpuUsage: 0,
+            userCpuUsage: 0,
+            sysCpuUsage: 0,
+            dataSet: {
+                totalMemKib: [],
+                anonMemKib: [],
+                fileMemKib: [],
+                sharedMemKib: [],
+                totalCpuUsage: [],
+                userCpuUsage: [],
+                sysCpuUsage: []
+            }
+        };
+    }
+}
+
+export interface ChanperfHistory {
```

Comment:
Let's call this `ChanperfEntry` or `ChanperfRecord`, or even just `Chanperf` because it represents a single response from the chanperf call.

Reply:
Since we need to send back all events found in the string I'm still leaning to it being a history object. The logic has been updated to only send back the events in the supplied string though. It no longer stores all the events for the session.